java/code/src/com/redhat/rhn/frontend/action/systems/SPMigrationAction.java

```diff
@@ -521,7 +521,10 @@ private String populateRequestFromAction(HttpServletRequest request, RequestCont
         List<Long> channelIds = channels.stream().map(Channel::getId).collect(Collectors.toList());
         List<EssentialChannelDto> childChannels = getChannelDTOs(ctx, baseChannel, channelIds);
 
-        SUSEProduct baseProduct = SUSEProductFactory.lookupByChannelName(baseChannel.getName()).get(0).getRootProduct();
+        // Get name of original base channel if channel is cloned
+        String origBaseChannelName = (baseChannel.isCloned() ? baseChannel.getOriginal() : baseChannel).getName();
```

Comment:
I think we might need to do some recursive lookup here, as the original can also be a clone :smile: 
(You can test, that it fails using 2 environments in the CLM project we discussed).
Maybe there is even some helper method somewhere in the code, I'll try to check that.

Reply:
You can give the `ChannelManager.getOriginalChannel`a shot!

Reply:
Indeed you were right. `Channel.getOriginal` only returns the direct predecessor.
`ChannelManager.getOriginalChannel` returns the original even in case of nested cloning.

TYVM for pointing this out. This problem did not come to my mind.